src/index.js

```diff
@@ -13,9 +15,15 @@ axios.get('/api/data')
         list.forEach( username => {
             contributors.push({
                 username,
-                mergedPRsNumber: data[username].mergedPRsNumber,
-                openPRsNumber: data[username].openPRsNumber,
-                issuesNumber: data[username].issuesNumber
+                mergedPRsNumber: data[username].mergedPRsCreatedTimes.filter((created_time)=> {
```

Comment:
@lolimay @RonLek I think on page load when there should be no filter. Whenever the user wants to filter then only the numbers should be updated according to the filter applied. Please do let me know your thoughts on this.

Are we implementing just filter by year or by specific start to end dates?

Reply:
I agree with @djcruz93 . We need to implement a date range picker in frontend so that we can select period dynamically instead of hardcoding the year in the code.

For example, GSoC 2021 actually starts on [October 26, 2020](https://developers.google.com/open-source/gsoc/timeline), perhaps admins need to select the range starts from October 26, 2020. It's a typical scenario  that we need to consider.

Reply:
@lolimay @djcruz93  I completely agree! As addressed in [this](https://github.com/RonLek/GSoC-Contribution-Leaderboard-Node/blob/2a6db85435d9f77936c7978260447460f6c6b4f6/src/index.js#L7-L8) comment it's only for the time-being until we have the frontend code ready which would change `year`'s value (we can rename the variable to `time`) when the admin fires an event by selecting a range from the filter UI on the frontend instead of hardcoding it the way it is now.

@lolimay we'll definitely do this by changing 

`${year}-01-01` to `${filter_start_time}...${filter_end_time}` 

[here](https://github.com/RonLek/GSoC-Contribution-Leaderboard-Node/blob/2a6db85435d9f77936c7978260447460f6c6b4f6/src/index.js#L82) as mentioned in this [article](https://docs.github.com/en/free-pro-team@latest/github/searching-for-information-on-github/searching-issues-and-pull-requests#search-by-when-an-issue-or-pull-request-was-created-or-last-updated) for ranges and 

`new Date(created_time).getFullYear() == year` to `filter_start_time <= ( new Date(created_time).getTime() )<= filter_end_time` 

[here](https://github.com/RonLek/GSoC-Contribution-Leaderboard-Node/blob/2a6db85435d9f77936c7978260447460f6c6b4f6/src/index.js#L84)  which I think would again depend on how we plan to get values from the frontend. 

The default values for `filter_start_time` can be `01-01-1970` and for `filter_end_time` can be `01-01-2100` which would give us no filter by default since all PRs and Issues would fall within that range. Let me know what you guys think of this idea.

Reply:
Nice! Let's go for it ;)